generated-sample-code/samplegen_map_field_access.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DO NOT EDIT! This is a generated sample ("Request",  "samplegen_map_field_access")
+
+// sample-metadata:
+//   title: This sample reads and loops over a map field in the response
+//   description: This sample reads and loops over a map field in the response
+//   usage: node samples/v1/samplegen_map_field_access.js
+
+'use strict';
+
+function main() {
+  // [START samplegen_map_field_access]
+
+  // Imports the client library
+  const {LanguageServiceClient} = require('@google-cloud/language').v1;
+
+  // Instantiates a client
+  const languageServiceClient = new LanguageServiceClient();
+
+  async function sampleAnalyzeEntities() {
+    const type = 'PLAIN_TEXT';
+    const language = 'en';
+
+    // The text content to analyze
+    const content =
+      'Googleplex is at 1600 Amphitheatre Parkway, Mountain View, CA.';
+    const document = {
+      type: type,
+      language: language,
+      content: content,
+    };
+
+    // Run request
+    const [response] = await languageServiceClient.analyzeEntities({
+      document: document,
+    });
+
+    for (const entity of response.entities) {
+      // Access value by key:
+      console.log(`URL: ${entity.metadata['wikipedia_url']}`);
+      // Loop over keys and values:
+      for (const [key, value] of Object.entries(entity.metadata)) {
+        console.log(`${key}: ${value}`);
+      }
+
+      // Loop over just keys:
+      for (const theKey of Object.keys(entity.metadata)) {
+        console.log(`Key: ${theKey}`);
+      }
+
+      // Loop over just values:
```

Comment:
My general feedback here is that we're not a JavaScript 101 course, we don't need to teach folks how to iterate collections. The `entries` example is probably enough.

Reply:
Ah. My intent: getting feedback on how the code looks (is it idiomatic?) when looping thru just the keys or just the values or, what will probably be the case most of the time, looping through the keys and values.

[Here is the configuration](https://github.com/beccasaurus/googleapis/commit/1d83780ab58c60e5e5573259eaa5c8138277e463):

```yaml
- comment: ["Access value by key:"]
- print:
  - "URL: %s"
  - map{"wikipedia_url"}

- comment: ["Loop over keys and values:"]
- loop:
    map: map
    key: key
    value: value
    body:
    - print: ["%s: %s", key, value]

- comment: ["Loop over just keys:"]
- loop:
    map: map
    key: the_key
    body:
    - print: ["Key: %s", the_key]

- comment: ["Loop over just values:"]
- loop:
    map: map
    value: the_value
    body:
    - print: ["Value: %s", the_value]
```

Reply:
Sample authors can access Maps in any of these ways in their configuration, so I'm showing all of the variations. 
e.g. maybe one of the variations which the generator renders isn't idiomatic

Reply:
( [full list of configs for these generated samples](https://github.com/beccasaurus/googleapis/compare/samples-for-syntax-review) )

Reply:
Ah, gotcha. I thought it will have all three in all cases.  Then all three look good!